importer/helpers/dagbuilder.go

```diff
@@ -43,14 +52,93 @@ func (dbp *DagBuilderParams) New(in <-chan []byte) *DagBuilderHelper {
 		ncb = nilFunc
 	}
 
-	return &DagBuilderHelper{
+	dbh := &DagBuilderHelper{
 		dserv:    dbp.Dagserv,
 		in:       in,
 		maxlinks: dbp.Maxlinks,
 		ncb:      ncb,
+		pipeline: make(chan *dag.Node),
+		Error:    make(chan error),
+	}
+	dbh.init()
```

Comment:
instead of this, move the entire construction into a 

```go
return NewDagBuilderHelper(dbp.Dagserv, in, dbp.Maxlinks, ncb)
```
where 
```go
func NewDagBuilderHelper(...) {
  db := &DagBuilderHelper{
    dserv:    dbp.Dagserv,        dserv:    dbp.Dagserv,
    in:       in,       in:       in,
    maxlinks: dbp.Maxlinks,       maxlinks: dbp.Maxlinks,
    ncb:      ncb,        ncb:      ncb,
    pipeline: make(chan *dag.Node),
    Error:    make(chan error),
  }

  hash := hasher()
  store := storeWorker(db.dserv)
  pin := pinner(db.ncb)

  hashedch := make(chan *dag.Node)
  storedch := make(chan *dag.Node)
  
  startWorkers(hash, hashWorkers, db.pipeline, hashedch, db.Error)
  startWorkers(store, storeWorkers, hashedch, storedch, db.Error)
  startWorkers(pin, pinWorkers, storedch, nil, db.Error)

  return db
}
```